swarm/pss/pss_test.go

```diff
@@ -61,429 +64,619 @@ func init() {
+		Addr: roaddr,
+	}
+	lclient.Call(nil, "pss_setRecipientAddressLength", 0)
+	err = lclient.Call(nil, "pss_sendAsym", hextopic, apimsg)
 	if err != nil {
-		t.Fatalf("exchange failed %v", err)
+		t.Fatalf("send fail: %v", err)
+	}
+	select {
+	case recvmsg := <-rmsgC:
+		if !bytes.Equal(recvmsg.Msg, apimsg.Msg) {
+			t.Fatalf("node 2 received payload mismatch: expected %v, got %v", apimsg.Msg, recvmsg.Msg)
+		}
+	case cerr := <-rctx.Done():
+		t.Fatalf("test message timed out: %v", cerr)
 	}
-}
-
-func TestSnapshot_50_5(t *testing.T) {
-	testSnapshot(t, "testdata/snapshot_50.json", 5, true)
-}
-
-func TestSnapshot_5_50(t *testing.T) {
-	testSnapshot(t, "testdata/snapshot_5.json", 50, true)
-}
 
-func TestSnapshot_5_5(t *testing.T) {
-	testSnapshot(t, "testdata/snapshot_5.json", 5, true)
 }
 
-func testSnapshot(t *testing.T, snapshotfile string, msgcount int, sim bool) {
-
+func TestCache(t *testing.T) {
+	var err error
+	var potaddr pot.Address
+	to, _ := hex.DecodeString("08090a0b0c0d0e0f1011121314150001020304050607161718191a1b1c1d1e1f")
+	keys, err := wapi.NewKeyPair()
+	privkey, err := w.GetPrivateKey(keys)
 
-	// choose the adapter to use
-	var adapter adapters.NodeAdapter
-	if sim {
-		adapter = adapters.NewSimAdapter(services)
-	} else {
-		baseDir, err := ioutil.TempDir("", "swarm-test")
-		if err != nil {
-			t.Fatal(err)
-		}
-		defer os.RemoveAll(baseDir)
-		adapter = adapters.NewExecAdapter(baseDir)
+	ps := NewTestPss(privkey, nil)
+	pp := NewPssParams(privkey)
+	data := []byte("foo")
+	datatwo := []byte("bar")
+	fwdaddr := network.RandomAddr()
+	copy(potaddr[:], fwdaddr.Over())
+	wparams := &whisper.MessageParams{
+		TTL:      DefaultTTL,
+		Src:      privkey,
+		Dst:      &privkey.PublicKey,
+		Topic:    PingTopic,
+		WorkTime: defaultWhisperWorkTime,
+		PoW:      defaultWhisperPoW,
+		Payload:  data,
+	}
+	woutmsg, err := whisper.NewSentMessage(wparams)
+	env, err := woutmsg.Wrap(wparams)
+	msg := &PssMsg{
+		Payload: env,
+		To:      to,
+	}
+	wparams.Payload = datatwo
+	woutmsg, err = whisper.NewSentMessage(wparams)
+	envtwo, err := woutmsg.Wrap(wparams)
+	msgtwo := &PssMsg{
+		Payload: envtwo,
+		To:      to,
 	}
 
-	// process shapshot
-	jsonsnapshot, err := ioutil.ReadFile(snapshotfile)
+	digest, err := ps.storeMsg(msg)
 	if err != nil {
-		t.Fatalf("cant read snapshot: %s", snapshotfile)
+		t.Fatalf("could not store cache msgone: %v", err)
 	}
-	snapshot := &simulations.Snapshot{}
-	err = json.Unmarshal(jsonsnapshot, snapshot)
+	digesttwo, err := ps.storeMsg(msgtwo)
 	if err != nil {
-		t.Fatalf("snapshot file unreadable: %v", err)
-	}
-	for _, node := range snapshot.Nodes {
-		node.Config.Services = []string{"bzz", "pss"}
+		t.Fatalf("could not store cache msgtwo: %v", err)
 	}
 
-	// setup network with snapshot
-	net := simulations.NewNetwork(adapter, &simulations.NetworkConfig{
-		ID: "0",
-	})
-	defer net.Shutdown()
+	if digest == digesttwo {
+		t.Fatalf("different msgs return same crc: %d", digesttwo)
```

Comment:
sorry it should say checksum. It checks if the checksum generated is getting different input. It's an old test, probably obsolete, maybe pointless.